server/parsec/components/invite.py

```diff
@@ -47,29 +47,16 @@
 
 @dataclass(slots=True)
 class BaseInvitationCreatedBy:
-    def for_invite_info(self) -> InviteInfoInvitationCreatedBy:
-        match self:
-            case InvitationCreatedByUser(user_id, human_handle):
-                return invited_cmds.latest.invite_info.InvitationCreatedByUser(
-                    user_id, human_handle
-                )
-            case InvitationCreatedByExternalService(service_label):
-                return invited_cmds.latest.invite_info.InvitationCreatedByExternalService(
-                    service_label
-                )
-            case unknown:
-                assert False, unknown
-
-    def for_invite_list(self) -> InviteListInvitationCreatedBy:
+    def cook_for_api[T](
```

Comment:
It's more of a question: but how does this work ? you pass the two callable everytime ?

Reply:
Yes pass the two callbacks and let the function choose what it needs

And as a matter of fact, the two callbacks are in fact just the classes that should be instantiated, but the typing is easier to express them as callbacks ^^